atomic_reactor/utils/resource_manager.py

```diff
@@ -6,193 +6,290 @@
-                    slot = self._read_slot_data(i)
-                    if slot == b'0\n':
-                        state = 'free'
-                        prid = ''
-                        self.free_slots.append(i)
-                    elif slot == "":
-                        logger.warning("host: %s slot: %s has invalid state", self.hostname, i)
-                        state = 'unknown'
-                        prid = 'unknown'
-                    else:
-                        state = 'running'
-                        slot_data = json.loads(slot)
-                        if self._validate_resource_lock(slot_data):
-                            prid = slot_data.get("prid", "unknown")
-                        else:
-                            logger.warning("host: %s slot: %s has invalid state", self.hostname, i)
-                            state = 'invalid'
-                            prid = 'invalid'
-                    logger.debug("%s slot %s is %s", meta['hostname'], i, state)
-                    self.slots.append({'sid': i, 'state': state, 'prid': prid})
-        except Exception as err:
-            logger.error(err)
-            logger.error("Availiability check failed fo host: %s", self.hostname)
-        return self.free_slots
-
-    def _write_free_slot(self, slot: int):
-        cmd = f'echo \'0\' > {ROOT_PATH}slot_{slot}.json'
-        data = self._ssh_run_remote_command(cmd)
-        return data
-
-    def unlock(self, slot: int, prid: str):
-        logger.debug('Unlocking')
-        retries = 0
-        while retries < 3:
-            ret = self._write_free_slot(slot)
-            if ret == b'':
-                logger.info('Unlocked')
-                return True
-            else:
-                logger.debug('Unlocking failed, retrying')
-                retries += 1
-        return False
-
-    def lock(self, slot: int, prid: str):
-        logger.debug("Aquiring lock")
-        payload = json.dumps({"prid": prid, "locked": str(datetime.now())})
-        retries = 0
-        while retries < 3:
-            filename = os.path.join(ROOT_PATH, f'slot_{slot}.json')
-            _cmd = 'if grep -qw 0 {0} ; then echo {1} > {0} ;else exit 1; fi'.format(
-                    filename,
-                    quote(payload)
-                    )
-            cmd = 'flock {} -c {}; echo $?'.format(filename, quote(_cmd))
-            logger.debug(cmd)
-            ret = self._ssh_run_remote_command(cmd)
-            if ret == b'0\n':
-                logger.info('Locked')
-                return True
-            else:
-                logger.debug('Locking failed, retrying')
-                retries += 1
-        logger.info('Locking failed')
-        raise Exception('Locking failed')
+            client.connect(
+                self.hostname, username=self.username, key_filename=self.ssh_keyfile)
+        except paramiko.ssh_exception.AuthenticationException as ex:
```

Comment:
Do we need this for each subclass?

Reply:
Dosn't backoff log something by itself anyway?

Also note that if you suppress the exceptions, backoff will not work

Reply:
Adam is right here. Logging can be added directly to backoff decorator and exception will break backoff completely. I made few tests here. https://gitlab.cee.redhat.com/-/snippets/4833 I'll make a fix.